### PR TITLE
feat(route): recover kaopu.news route

### PR DESCRIPTION
**Pull request by fashioncj**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/kaopu/news/zh-hans
/kaopu/news/zh-hant
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
recover & refactor route from deprecate route

### CODE REVIEW

File: lib/routes/kaopunews/namespace.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import type { Namespace } from '@/types';
+
+export const namespace: Namespace = {
+    name: '靠谱新闻',
+    url: 'https://kaopu.news',
```
Comment (TonyRL) [inline lib/routes/kaopunews/namespace.ts:5]:
```suggestion
    url: 'kaopu.news',
```
> url: The website URL **without protocol** that corresponds

docs: https://docs.rsshub.app/joinus/new-rss/start-code

File: lib/routes/kaopunews/news.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/news/:language?',
+    categories: ['programming'],
```
Comment (TonyRL) [inline lib/routes/kaopunews/news.ts:7]: This does not belong to `programming`

File: lib/routes/kaopunews/news.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    example: '/news/zh-hans',
+    parameters: {
+        language: 'zh-hans | zh-hant',
```
Comment (TonyRL) [inline lib/routes/kaopunews/news.ts:10]: Please provide more concise explanations.

File: lib/routes/kaopunews/news.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    radar: [
+        {
+            source: ['kaopu.news'],
```
Comment (TonyRL) [inline lib/routes/kaopunews/news.ts:14]:
```suggestion
            source: ['kaopu.news/'],
```

File: lib/routes/kaopunews/news.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        },
+    ],
+    name: '靠谱新闻',
```
Comment (TonyRL) [inline lib/routes/kaopunews/news.ts:17]: `Route.name` should not be the same as `Namespace.name`
https://github.com/DIYgod/RSSHub/blob/9a2e50ef09815a44cf555fe686b1ba48bb1735ef/lib/routes/kaopunews/namespace.ts#L4

File: lib/routes-deprecated/kaopunews/index.js [hunk 6]
```diff
@@ -1,28 +0,0 @@
-const got = require('@/utils/got');
```
Comment (TonyRL) [inline lib/routes-deprecated/kaopunews/index.js:1]: Please remove this entry as well.
https://github.com/DIYgod/RSSHub/blob/cb1ac8e0b4936b8b24943a5689606c0480c37cf8/lib/router.js#L1272

**Review by TonyRL — COMMENTED** [review]
Please use `kaopu` instead
https://docs.rsshub.app/joinus/new-rss/start-code#creating-namespace

File: lib/routes/kaopunews/namespace.ts (hunk 1 above)
Comment (fashioncj) [reply lib/routes/kaopunews/namespace.ts]: fix

File: lib/routes/kaopu/news.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    example: '/news/zh-hans, /news/zh-hant',
+    parameters: {
+        language: 'zh-hans | zh-hant',
+    },
+    radar: [
+        {
+            source: ['kaopu.news/'],
+        },
+    ],
+    name: '靠谱新闻-全部',
```
Comment (TonyRL) [inline lib/routes/kaopu/news.ts:17]:
```suggestion
    name: '全部',
```

File: lib/routes/kaopu/news.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import { Route } from '@/types';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/news/:language?',
+    categories: ['new-media'],
+    example: '/news/zh-hans, /news/zh-hant',
+    parameters: {
+        language: 'zh-hans | zh-hant',
```
Comment (TonyRL) [inline lib/routes/kaopu/news.ts:10]: https://github.com/DIYgod/RSSHub/pull/14818#discussion_r1527605841

**Pull request merged**